WeakAuras/Prototypes.lua

```diff
@@ -11244,9 +11288,160 @@ Private.event_prototypes = {
+            active = false
+            if attribute ~= "quantity" and attribute ~= "realCharacterQuantity" then
+              primaryCheckFailed = true
+              break;
+            end
+          end
+        end
+
+        -- Helper function to check tristate values
+        local function checkTristate(tristateValue, attribute, condition, primary)
+          if active and tristateValue ~= "nil" and condition then
+            if tostring(attribute) ~= tristateValue then
+              active = false
+              primaryCheckFailed = primary and true
+            end
+          end
+        end
+
+        -- Check tristate values
+        checkTristate(discovered, currencyInfo.discovered, true, true)
+        checkTristate(capped, currencyInfo.quantity >= (currencyInfo.maxQuantity or 0), currencyInfo.maxQuantity and currencyInfo.maxQuantity > 0, false)
+        checkTristate(seasonCapped, currencyInfo.totalEarned >= (currencyInfo.maxQuantity or 0), currencyInfo.useTotalEarnedForMaxQty and currencyInfo.maxQuantity and currencyInfo.maxQuantity > 0, true)
+        checkTristate(weeklyCapped, currencyInfo.quantityEarnedThisWeek >= (currencyInfo.maxWeeklyQuantity or 0), currencyInfo.maxWeeklyQuantity and currencyInfo.maxWeeklyQuantity > 0, true)
+
+        local sharedStateValues = {
+          name = currencyInfo.name,
+          icon = currencyInfo.iconFileID,
+          total = currencyInfo.maxQuantity,
+          maxQuantity = currencyInfo.maxQuantity,
+          progressType = "static",
+          accountQuantity = currencyInfo.accountQuantity,
+          realAccountQuantity = currencyInfo.realAccountQuantity,
+          description = currencyInfo.description,
+          quality = currencyInfo.quality,
+          discovered = currencyInfo.discovered,
+          transferPercentage = currencyInfo.transferPercentage,
+          quantityEarnedThisWeek = currencyInfo.quantityEarnedThisWeek,
+          totalEarned = currencyInfo.totalEarned,
+          changed = true,
+        }
+
+        states[""] = states[""] or {}
+        local state = states[""]
+        state.show = active
+        state.value = currencyInfo.quantity
+        state.realCharacterQuantity = currencyInfo.realCharacterQuantity
+        state.capped = currencyInfo.maxQuantity and currencyInfo.maxQuantity > 0 and currencyInfo.quantity >= currencyInfo.maxQuantity
+        state.seasonCapped = currencyInfo.useTotalEarnedForMaxQty and currencyInfo.maxQuantity and currencyInfo.maxQuantity > 0 and currencyInfo.totalEarned >= currencyInfo.maxQuantity
+        state.weeklyCapped = currencyInfo.maxWeeklyQuantity and currencyInfo.maxWeeklyQuantity > 0 and currencyInfo.quantityEarnedThisweek >= currencyInfo.maxWeeklyQuantity
+        state.characterName = UnitName("player")
+        state.characterGUID = UnitGUID("player")
+        state.mainCharacter = true
+
+        if clone then
+          for i, currencyData in ipairs(accountCurrencyData) do
```

Comment:
Transfering **everything** from a character doesn't correctly update the trigger, the clone for the character who know has 0 is still displayed.

I think this because this loop here runs over accountCurrencyData which no longer contains any data for that character, so the existing state is kept unmodified even though it should have been dropped.